test/dynamics/test_driver.py

```diff
@@ -240,3 +240,22 @@ def test_run_twice():
     driver.run(0.03)
     driver.run(0.03)
     np.testing.assert_allclose(driver.t, 0.06)
+
+
+def test_change_integrator():
+    ha, vstate, _ = _setup_system(L=2)
+    driver = nkx.TDVP(
+        ha,
+        vstate,
+        nkx.dynamics.RK23(dt=0.01, adaptive=False),
+    )
+    driver.run(0.03)
+    np.testing.assert_allclose(driver.t, 0.03)
+
+    integrator = nkx.dynamics.Euler(dt=0.05)
+    driver.integrator = integrator
+    np.testing.assert_allclose(driver.t, 0.03)
+    np.testing.assert_allclose(driver.dt, 0.05)
+
+    driver.run(0.1)
+    np.testing.assert_allclose(driver.t, 0.13)
```

Comment:
I think it would be good to add a few lines testing a change of the error norm too.